src/WrappedIterator.cpp

```diff
@@ -0,0 +1,33 @@
+//  Copyright (c) 2016, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+
+#include "iterlib/WrappedIterator.h"
+
+namespace iterlib {
+
+folly::Future<folly::Unit> WrappedIterator::prepare() {
+  if (prepared_) {
+    return folly::makeFuture();
+  }
+  if (innerIter_) {
+    return innerIter_->prepare()
+      .then([this]() {
+          // Code to extract any data from inner iter
```

Comment:
Indentation. Lambda code block is indented +2 compared to outer scope (this is +4).